flink-java/src/main/java/org/apache/flink/api/java/operators/SingleInputUdfOperator.java

```diff
@@ -54,8 +54,11 @@
 
 	private Map<String, DataSet<?>> broadcastVariables;
 
+	// NOTE: only set this variable via setSemanticProperties()
```

Comment:
I think this refactoring is quite fragile. The semantic properties utility is not returning an empty properties object, but null and you take care of setting it correctly here depending on whether the forwarded fields have been set manually or not.

If optimize is enabled and there are manual annotations, they will be overriden. I am wondering if it is better to have manual annotations trump optimizer annotations. What's your opinion on this?